tests/test_gossip.py

```diff
@@ -1036,10 +1033,7 @@ def test_gossip_addresses(node_factory, bitcoind):
     nodes = l2.rpc.listnodes(l1.info['id'])['nodes']
     assert len(nodes) == 1 and nodes[0]['addresses'] == [
         {'type': 'ipv4', 'address': '127.0.0.1', 'port': 2},
-        {'type': 'ipv4', 'address': '127.0.0.1', 'port': 9735},
         {'type': 'ipv6', 'address': '::', 'port': 3},
-        {'type': 'ipv6', 'address': '::', 'port': 9735},
-        {'type': 'torv3', 'address': 'vww6ybal4bd7szmgncyruucpgfkqahzddi37ktceo3ah7ngmcopnpyyd.onion', 'port': 9735},
```

Comment:
Mh, I did not catch why you are removing the test here.

Reply:
Because this was caused by lines with address and no port a couple of lines earlier:

```diff
@@ -1018,10 +1018,7 @@ def test_gossip_addresses(node_factory, bitcoind):
     l1 = node_factory.get_node(options={
         'announce-addr': [
             '[::]:3',
-            '[::]',
             '127.0.0.1:2',
-            '127.0.0.1',
-            'vww6ybal4bd7szmgncyruucpgfkqahzddi37ktceo3ah7ngmcopnpyyd.onion',
             '4acth47i6kxnvkewtm6q7ib2s3ufpo5sqbsnzjpbi7utijcltosqemad.onion:123
4'
         ],
     })
```

And since each different network would get a different default port when the port is not specified, I have removed all these default-port lines. I think it makes sense to not replicate the same logic which already is in `lightningd.c` into the Python test since it would just be repetition IMHO.

Reply:
Mh this can introduce some regression in the future. I think introducing a duplication logic in the test is fine because we are testing if the code is really working as expected.

I like the idea to have this part of the code covered by a integration test, and maybe this can be a solution https://github.com/ElementsProject/lightning/pull/4900#discussion_r761823282

Reply:
https://github.com/ElementsProject/lightning/pull/4958